graylog2-server/src/main/java/org/graylog2/bootstrap/ServerBootstrap.java

```diff
@@ -180,9 +181,9 @@ private void runPreflightWeb(List<Module> preflightCheckModules) {
     }
 
     private void doRunWithPreflightInjector(Injector preflightInjector) {
-        final PreflightConfigService preflightConfigService = preflightInjector.getInstance(PreflightConfigService.class);
+        final PreflightBoot preflightBoot = preflightInjector.getInstance(PreflightBoot.class);
 
-        final Supplier<Boolean> shouldRun = () -> preflightConfigService.getPersistedConfig().isEmpty() || configuration.enablePreflightWebserver();
+        final Supplier<Boolean> shouldRun = preflightBoot::shouldRunPreflightWeb;
 
         if (!shouldRun.get()) {
```

Comment:
I believe you do not need `Supplier` now, you can just call `shouldRunPreflightWeb` in the if-clause.